[SVG2] Remove requiredFeatures from SVGTests

Ahmad-S792 opened the pull request:
#### 4fb2feec1fbe720c538bda20dfe6efde401b757e
<pre>
[SVG2] Remove requiredFeatures from SVGTests

<a href="https://bugs.webkit.org/show_bug.cgi?id=204125">https://bugs.webkit.org/show_bug.cgi?id=204125</a>
<a href="https://rdar.apple.com/123784300">rdar://123784300</a>

Reviewed by Simon Fraser.

This patch aligns WebKit with Gecko / Firefox, Blink / Chromium.

It was removed from web specification [1] &amp; [2] in 2015.

[1] <a href="https://github.com/w3c/svgwg/commit/9a30d01f6410dc516c5f874d71e957230a3448cd">https://github.com/w3c/svgwg/commit/9a30d01f6410dc516c5f874d71e957230a3448cd</a>
[2] <a href="https://www.w3.org/2015/06/09-svg-irc#T15-35-40">https://www.w3.org/2015/06/09-svg-irc#T15-35-40</a>

Just to add, it was removed from Blink in 2017 [3]:

[3] <a href="https://chromium.googlesource.com/chromium/src.git/+/3bc8c04a9e2e711079299a591fa9eadc56b09729">https://chromium.googlesource.com/chromium/src.git/+/3bc8c04a9e2e711079299a591fa9eadc56b09729</a>

Credits to Said for initial patch.

* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-04-f-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-04-f-manual-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-04-f-manual-expected.txt:
* LayoutTests/imported/mozilla/svg/svg-integration/dynamic-conditions-outer-svg-03.xhtml:
* LayoutTests/imported/mozilla/svg/switch-01.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/historical-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-04-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/types-dom-06-f-manual.svg:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual-expected.txt:
* LayoutTests/imported/mozilla/svg/conditions-01.svg:
* LayoutTests/imported/mozilla/svg/conditions-06-expected.svg: Removed.
* LayoutTests/imported/mozilla/svg/conditions-06.svg: Removed.
* LayoutTests/imported/mozilla/svg/dynamic-conditions-01.svg:
* LayoutTests/imported/mozilla/svg/svg-integration/dynamic-conditions-outer-svg-03.xhtml:
* LayoutTests/imported/mozilla/svg/switch-01.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/historical-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-04-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/types-dom-06-f-manual.svg:
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGCircleElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGCircleElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGEllipseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGEllipseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGForeignObjectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGForeignObjectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGImageElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGImageElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGLineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGLineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPathElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPathElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPolygonElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPolygonElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPolylineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPolylineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGRectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGRectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGSVGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGSVGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGTextElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGTextElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGUseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGUseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGCircleElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGCircleElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGEllipseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGEllipseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGForeignObjectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGForeignObjectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGImageElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGImageElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGLineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGLineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPathElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPathElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPolygonElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPolygonElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPolylineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPolylineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGRectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGRectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGSVGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGSVGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGTextElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGTextElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGUseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGUseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual-expected.txt:
* LayoutTests/platform/mac/svg/dynamic-updates/SVGCircleElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGCircleElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGEllipseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGEllipseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGForeignObjectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGForeignObjectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGImageElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGImageElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGLineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGLineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPathElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPathElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPolygonElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPolygonElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPolylineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPolylineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGRectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGRectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGSVGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGSVGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGTextElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGTextElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGUseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGUseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/svg/custom/outer-svg-unknown-feature-expected.txt: Removed.
* LayoutTests/svg/custom/outer-svg-unknown-feature.svg: Removed.
* LayoutTests/svg/dom/SVGStringList-basics-expected.txt:
* LayoutTests/svg/dom/SVGStringList-basics.xhtml:
* LayoutTests/svg/dom/SVGStringList-expected.txt:
* LayoutTests/svg/dom/SVGStringList.html:
* LayoutTests/svg/dom/SVGTests-expected.txt:
* LayoutTests/svg/dom/SVGTests.html:
* LayoutTests/svg/dynamic-updates/SVGCircleElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGCircleElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGCircleElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGCircleElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGEllipseElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGEllipseElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGEllipseElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGEllipseElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGForeignObjectElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGForeignObjectElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGForeignObjectElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGForeignObjectElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGGElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGGElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGGElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGGElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGImageElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGImageElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGImageElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGImageElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGLineElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGLineElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGLineElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGLineElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPathElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPathElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPathElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPathElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolygonElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolygonElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolygonElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolygonElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolylineElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolylineElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolylineElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolylineElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGRectElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGRectElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGRectElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGRectElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGSVGElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGSVGElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGSVGElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGSVGElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGTextElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGTextElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGTextElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGTextElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGUseElement-dom-href1-attr.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGUseElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGUseElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGUseElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGUseElement-svgdom-requiredFeatures.html: Removed.
* Source/WebCore/svg/SVGTests.cpp:
(WebCore::SVGConditionalProcessingAttributes::SVGConditionalProcessingAttributes):
(WebCore::SVGTests::hasExtension):
(WebCore::SVGTests::isValid const):
(WebCore::SVGTests::parseAttribute):
(WebCore::SVGTests::addSupportedAttributes):
(): Deleted.
(WebCore::SVGTests::protectedRequiredFeatures): Deleted.
(WebCore::SVGTests::hasFeatureForLegacyBindings): Deleted.
* Source/WebCore/svg/SVGTests.h:
(WebCore::SVGConditionalProcessingAttributes::requiredFeatures): Deleted.
(WebCore::SVGTests::requiredFeatures): Deleted.
* Source/WebCore/svg/SVGTests.idl:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDOMImplementation.cpp:
(webkit_dom_dom_implementation_has_feature):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMNodeGtk.cpp:
(webkit_dom_node_is_supported):
* Source/WebKitLegacy/mac/DOM/DOMImplementation.mm:
(-[DOMImplementation hasFeature:version:]):
* Source/WebKitLegacy/mac/DOM/DOMNode.mm:
(-[DOMNode isSupported:version:]):

Canonical link: <a href="https://commits.webkit.org/282754@main">https://commits.webkit.org/282754@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e33b21f45a92e8cc4b1e9d2efee26998e364e14

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/64130 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/43487 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16727 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/68152 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14738 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/66250 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/51185 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/15018 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51645 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10180 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/67199 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/40239 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55497 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32264 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36911 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12880 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13612 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/58872 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/13209 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69851 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/8077 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12737 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58964 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/8110 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55593 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/59121 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14171 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6693 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/39307 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/40386 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/41569 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/40129 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKitLegacy/mac/DOM/DOMNode.mm:
```diff
@@ -273,7 +273,7 @@ - (void)normalize
 - (BOOL)isSupported:(NSString *)feature version:(NSString *)version
 {
-    return SVGTests::hasFeatureForLegacyBindings(feature, version);
+    return true;
```
We're in Objective-C, so this should `return YES`

Ahmad-S792 replied:
Done in https://github.com/WebKit/WebKit/commit/8e33b21f45a92e8cc4b1e9d2efee26998e364e14.

smfr commented on Source/WebKitLegacy/mac/DOM/DOMImplementation.mm:
```diff
@@ -59,7 +59,7 @@ - (void)dealloc
 - (BOOL)hasFeature:(NSString *)feature version:(NSString *)version
 {
-    return WebCore::SVGTests::hasFeatureForLegacyBindings(feature, version);
+    return true;
```
We're in Objective-C, so this should `return YES`

Ahmad-S792 replied:
Done in https://github.com/WebKit/WebKit/commit/8e33b21f45a92e8cc4b1e9d2efee26998e364e14.

Ahmad-S792 committed:
[SVG2] Remove requiredFeatures from SVGTests

https://bugs.webkit.org/show_bug.cgi?id=204125
rdar://123784300

Reviewed by Simon Fraser.

This patch aligns WebKit with Gecko / Firefox, Blink / Chromium.

It was removed from web specification [1] & [2] in 2015.

[1] https://github.com/w3c/svgwg/commit/9a30d01f6410dc516c5f874d71e957230a3448cd
[2] https://www.w3.org/2015/06/09-svg-irc#T15-35-40

Just to add, it was removed from Blink in 2017 [3]:

[3] https://chromium.googlesource.com/chromium/src.git/+/3bc8c04a9e2e711079299a591fa9eadc56b09729

Credits to Said for initial patch.

* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-04-f-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-04-f-manual-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-04-f-manual-expected.txt:
* LayoutTests/imported/mozilla/svg/svg-integration/dynamic-conditions-outer-svg-03.xhtml:
* LayoutTests/imported/mozilla/svg/switch-01.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/historical-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-04-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/types-dom-06-f-manual.svg:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual-expected.txt:
* LayoutTests/imported/mozilla/svg/conditions-01.svg:
* LayoutTests/imported/mozilla/svg/conditions-06-expected.svg: Removed.
* LayoutTests/imported/mozilla/svg/conditions-06.svg: Removed.
* LayoutTests/imported/mozilla/svg/dynamic-conditions-01.svg:
* LayoutTests/imported/mozilla/svg/svg-integration/dynamic-conditions-outer-svg-03.xhtml:
* LayoutTests/imported/mozilla/svg/switch-01.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/historical-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-04-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/import/types-dom-06-f-manual.svg:
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGCircleElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGCircleElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGEllipseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGEllipseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGForeignObjectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGForeignObjectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGImageElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGImageElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGLineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGLineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPathElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPathElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPolygonElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPolygonElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPolylineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGPolylineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGRectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGRectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGSVGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGSVGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGTextElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGTextElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGUseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/gtk/svg/dynamic-updates/SVGUseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGCircleElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGCircleElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGEllipseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGEllipseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGForeignObjectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGForeignObjectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGImageElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGImageElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGLineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGLineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPathElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPathElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPolygonElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPolygonElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPolylineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGPolylineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGRectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGRectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGSVGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGSVGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGTextElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGTextElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGUseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac-sonoma-wk2-pixel/svg/dynamic-updates/SVGUseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-02-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-03-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/struct-cond-overview-05-f-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/svgdom-over-01-f-manual-expected.txt:
* LayoutTests/platform/mac/svg/dynamic-updates/SVGCircleElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGCircleElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGEllipseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGEllipseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGForeignObjectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGForeignObjectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGImageElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGImageElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGLineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGLineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPathElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPathElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPolygonElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPolygonElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPolylineElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGPolylineElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGRectElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGRectElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGSVGElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGSVGElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGTextElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGTextElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGUseElement-dom-requiredFeatures-expected.png: Removed.
* LayoutTests/platform/mac/svg/dynamic-updates/SVGUseElement-svgdom-requiredFeatures-expected.png: Removed.
* LayoutTests/svg/custom/outer-svg-unknown-feature-expected.txt: Removed.
* LayoutTests/svg/custom/outer-svg-unknown-feature.svg: Removed.
* LayoutTests/svg/dom/SVGStringList-basics-expected.txt:
* LayoutTests/svg/dom/SVGStringList-basics.xhtml:
* LayoutTests/svg/dom/SVGStringList-expected.txt:
* LayoutTests/svg/dom/SVGStringList.html:
* LayoutTests/svg/dom/SVGTests-expected.txt:
* LayoutTests/svg/dom/SVGTests.html:
* LayoutTests/svg/dynamic-updates/SVGCircleElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGCircleElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGCircleElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGCircleElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGEllipseElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGEllipseElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGEllipseElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGEllipseElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGForeignObjectElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGForeignObjectElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGForeignObjectElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGForeignObjectElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGGElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGGElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGGElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGGElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGImageElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGImageElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGImageElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGImageElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGLineElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGLineElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGLineElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGLineElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPathElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPathElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPathElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPathElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolygonElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolygonElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolygonElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolygonElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolylineElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolylineElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolylineElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGPolylineElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGRectElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGRectElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGRectElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGRectElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGSVGElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGSVGElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGSVGElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGSVGElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGTextElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGTextElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGTextElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGTextElement-svgdom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGUseElement-dom-href1-attr.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGUseElement-dom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGUseElement-dom-requiredFeatures.html: Removed.
* LayoutTests/svg/dynamic-updates/SVGUseElement-svgdom-requiredFeatures-expected.txt: Removed.
* LayoutTests/svg/dynamic-updates/SVGUseElement-svgdom-requiredFeatures.html: Removed.
* Source/WebCore/svg/SVGTests.cpp:
(WebCore::SVGConditionalProcessingAttributes::SVGConditionalProcessingAttributes):
(WebCore::SVGTests::hasExtension):
(WebCore::SVGTests::isValid const):
(WebCore::SVGTests::parseAttribute):
(WebCore::SVGTests::addSupportedAttributes):
(): Deleted.
(WebCore::SVGTests::protectedRequiredFeatures): Deleted.
(WebCore::SVGTests::hasFeatureForLegacyBindings): Deleted.
* Source/WebCore/svg/SVGTests.h:
(WebCore::SVGConditionalProcessingAttributes::requiredFeatures): Deleted.
(WebCore::SVGTests::requiredFeatures): Deleted.
* Source/WebCore/svg/SVGTests.idl:
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDOMImplementation.cpp:
(webkit_dom_dom_implementation_has_feature):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMNodeGtk.cpp:
(webkit_dom_node_is_supported):
* Source/WebKitLegacy/mac/DOM/DOMImplementation.mm:
(-[DOMImplementation hasFeature:version:]):
* Source/WebKitLegacy/mac/DOM/DOMNode.mm:
(-[DOMNode isSupported:version:]):

Canonical link: https://commits.webkit.org/282754@main
